src/asynqp/channel.py

```diff
@@ -65,14 +66,16 @@ def declare_exchange(self, name, type, *, durable=True, auto_delete=False, inter
                              "Valid names consist of letters, digits, hyphen, underscore, period, or colon, "
                              "and do not begin with 'amq.'")
 
-        self.sender.send_ExchangeDeclare(name, type, durable, auto_delete, internal, arguments or {})
-        yield from self.synchroniser.await(spec.ExchangeDeclareOK)
+        self.sender.send_ExchangeDeclare(
+            name, type, durable, auto_delete, internal, nowait, arguments or {})
+        if not nowait:
+            yield from self.synchroniser.await(spec.ExchangeDeclareOK)
+            self.reader.ready()
```

Comment:
ready() means, that we already process this frame and ready to receive next one. with nowait DeclareOK will not be sent, so we did not process any frame, no need for `ready()`